arangod/Replication2/ReplicatedLog/ILogInterfaces.h

```diff
@@ -75,12 +76,38 @@ struct ILogParticipant {
   using WaitForQueue = std::multimap<LogIndex, WaitForPromise>;
 
   [[nodiscard]] virtual auto waitFor(LogIndex index) -> WaitForFuture = 0;
-  [[nodiscard]] virtual auto waitForIterator(LogIndex index) -> WaitForIteratorFuture;
+  [[nodiscard]] virtual auto waitForIterator(LogIndex index) -> WaitForIteratorFuture = 0;
   [[nodiscard]] virtual auto getTerm() const noexcept -> std::optional<LogTerm>;
+  [[nodiscard]] virtual auto getCommitIndex() const noexcept -> LogIndex = 0;
 
   [[nodiscard]] virtual auto release(LogIndex doneWithIdx) -> Result = 0;
 };
 
+/**
+ * Interface describing a LogFollower API. Components should use this interface
+ * if they want to refer to a LogFollower instance.
+ */
+struct ILogFollower : ILogParticipant, AbstractFollower {
+  virtual auto waitForLeaderAcked() -> WaitForFuture = 0;
+};
```

Comment:
We could do that, but I like the clear distinction that a Leader only needs something where it can call `appendEntries`. In particular, this would require us to make the `LocalFollower´ a full blown follower that supports `waitFor` operations.